tox.ini

```diff
@@ -77,3 +77,24 @@ commands =
       --source={[vars]src_path},{[vars]lib_path} \
       -m pytest -v --tb native {posargs} {[vars]tst_path}/unit
     coverage report
+
+[testenv:integration]
+description = Run integration tests
+deps =
+    git+https://github.com/juju/python-libjuju.git
+    pytest
+    git+https://github.com/charmed-kubernetes/pytest-operator.git
+commands =
+    pytest -v --tb native --log-cli-level=INFO -s {posargs} {toxinidir}/tests/integration
+
+[testenv:integration-lma]
```

Comment:
I don't know how big the lma-light or lma-ha bundles are going to get, but you may eventually not be able to run the full integration tests in this repo. The Charmed Kubeflow bundle is big enough that it costs us money to run the full and lite test suites because we have to spin up an external VM, so we only run it in the Charmed Kubeflow CI. One of the sub-bundles in it is Kubeflow Pipelines, which by itself almost hits the Github Actions runner resource limits, and doesn't actually have all that many charms in it:

https://github.com/canonical/kfp-operators/blob/master/bundle.yaml

If you did pull bundle-level integration testing out of this repo, one thing you could do to mitigate the impact is have CI for the bundle run on a schedule, like we do for testing Charmed Kubeflow on AKS:

https://github.com/canonical/bundle-kubeflow/blob/master/.github/workflows/test-aks.yaml

Which you may want to do anyways, as that will catch any dependencies like Juju or MicroK8s changing unexpectedly, which is an issue we've had for the Charmed Kubeflow bundle. Since most of the PRs and general activity happen in repos for sub-bundles like https://github.com/canonical/kfp-operators/, the CI for https://github.com/canonical/bundle-kubeflow/ is run irregularly, and we've been meaning to put in place more scheduled CI there as well.

Reply:
The intention here is to be able to run integration tests both locally and as part of CI.

> If you did pull bundle-level integration testing out of this repo, one thing you could do to mitigate the impact is have CI for the bundle run on a schedule

Good point. Although if the lma-bundle repo is the one that would be running tests on a schedule, it would still need to checkout the main branch of this repo, build the charm etc., so I'm not sure if it is advantageous. I think that running tests with obvious/correct context has value/increases merge responsibility.
Something in juju/microk8s etc changing is a good use case though. I will add it to the bundle repo shortly.
Some of such surprises could be mitigated by version freezing of the pypi and oci images.